testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/AbstractSSLMasterSlaveTestCase.java

```diff
@@ -0,0 +1,188 @@
+        ModelNode operation = createOpNode("host=master/core-service=management/management-interface=native-interface",
+                ModelDescriptionConstants.WRITE_ATTRIBUTE_OPERATION);
+        operation.get("name").set("security-realm");
+        operation.get("value").set("ManagementRealm");
+        executeOverHttp(getMasterMgmtUri(), operation.toJSONString(true));
+    }
+
+    protected static void reloadMaster() throws Exception {
+        ModelNode op = new ModelNode();
+        op.get(OP).set("reload");
+        op.get(OP_ADDR).add(HOST, "master");
+        op.get(ADMIN_ONLY).set(false);
+        executeOverHttp(getMasterMgmtUri(), op.toJSONString(true));
+    }
+
+    protected static void checkHostStatusOnMaster(String host) throws Exception {
+        final URI mgmtURI = getMasterMgmtUri();
+        final String operation = getHostStateRunningOperationJson(host);
+        final long time = System.currentTimeMillis() + TIMEOUT;
+        do {
+            Thread.sleep(1000);
+            if (isHostStateRunning(mgmtURI, operation)) {
+                return;
+            }
+        } while (System.currentTimeMillis() < time);
+
+        Assert.fail("Cannot validate host '" + host + "' is running");
+    }
+
+    private static URI getMasterMgmtUri() throws MalformedURLException, URISyntaxException {
+        return new URL("http", DomainTestSupport.masterAddress, MANAGEMENT_HTTP_PORT, MGMT_CTX).toURI();
+    }
+
+    private static boolean isHostStateRunning(URI mgmtURI, String operation) throws IOException {
+        ModelNode responseNode = executeOverHttp(mgmtURI, operation);
+        if (responseNode != null && responseNode.get(OUTCOME).asString().equals(SUCCESS)) {
+            final ModelNode resultNode = responseNode.require(RESULT);
+            if ("running".equalsIgnoreCase(resultNode.asString())) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private static ModelNode executeOverHttp(URI mgmtURI, String operation) throws IOException {
+        CloseableHttpClient httpClient = getHttpClient(mgmtURI);
+        HttpEntity operationEntity = new StringEntity(operation, ContentType.APPLICATION_JSON);
+        HttpPost httpPost = new HttpPost(mgmtURI);
+        httpPost.setEntity(operationEntity);
+
+        HttpResponse response;
+        ModelNode responseNode;
+        try{
+            response = httpClient.execute(httpPost);
+
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (statusCode != 200) {
+                return null;
+            }
+
+            HttpEntity entity = response.getEntity();
+            if (entity == null) {
+                return null;
+            }
+            responseNode = ModelNode.fromJSONStream(response.getEntity().getContent());
+            EntityUtils.consume(entity);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+            return null;
```

Comment:
Why over HTTP? Because I do not know how to use native interface with SSL here.
Or, another why?